README.md

```diff
@@ -31,9 +31,22 @@ So, for example, to install Elixir 1.5.0 and take advantage of the new features
 Be sure to also install the corresponding Erlang/OTP version with asdf-erlang, and to have both selected versions in your
 `.tool-versions` file.
 
-## Compiling from a git reference
+## Compiling from a git reference or source
 
-You can compile your own elixir from a commit reference from the [elixir github repository](https://github.com/elixir-lang/elixir/commits/master).
+You can download and compile elixir as of a specific commit reference from the [elixir github repository](https://github.com/elixir-lang/elixir/commits/master). For example:
+
+```
+asdf install elixir ref:161f5f9a3b6bc38faa152e08c567486b71b0602a
+```
+
+You can also [compile elixir from source](https://github.com/elixir-lang/elixir/tree/master#compiling-from-source) without using `asdf` (for example, so that you can use the `master` branch of elixir or a branch with your own modifications), then symlink and use it as a named version:
+
+```
+ln -s /path/to/elixir ~/.asdf/installs/elixir/master
+asdf reshim elixir master
```

Comment:
This is a hack and is not the recommended way of doing things. You should instead use the `path:<path>` version format to specify that you want to use your own version of Elixir https://asdf-vm.com/#/core-configuration?id=tool-versions Doing this will still allow you to checkout any version of Elixir and recompile on the fly without having to reconfigure asdf.

Reply:
It is indeed a hack but I don't think you have to reconfigure asdf if you change version or recompile elixir?

Reply:
Agreed, plus `asdf local elixir master` is much cleaner than `asdf local elixir path:/let/me/remember/the/path/to/my/elixir/repo`

Reply:
With either approach (linking as shown here, or using the `path:<path>` syntax in your `.tool-versions` file) you will be free to checkout different refs and recompile Elixir without having to change anything else.

Altering anything inside your asdf data dir (`~/.asdf/`) is not recommended. asdf is simple enough that it will recognize the `master` directory if you create it, but it's ultimately hack. If you try to uninstall `master` asdf will probably try to delete whatever you have in your `/path/to/elixir` (I've not tried this, but I don't think we guard against this).

Reply:
@TheFirstAvenger you shouldn't need to run that command very often. Once you've got the version set for your project you shouldn't ever need to run that command again. If you find yourself needing to use that version in a lot of different projects, you can set that as the global version, or define an shell alias for that `asdf local` command to make it easier to run.

Reply:
There is also `ref:master`, but I don't think that is what you want here - https://www.amberbit.com/blog/2017/10/30/trying-out-upcoming-elixir-releases/